### PR TITLE
[skip ci] Initial artifactory verification test

### LINKED ISSUE #6602: Write docker registry validation for login and pull against Artifactory
As a VIC user, I expect VIC to work against Artifactory's docker registry.

Acceptance Criteria:
- [ ]  Nightly validation tests to verify Artifactory's compatibility

---
Issue was resolved in #5338.  We need validation scripts.

### PR DESCRIPTION
**Pull request by mhagen-vmware**
fixes #6602 

Still hitting this issue though on the image pull:
```Error response from daemon: Failed to obtain OAuth endpoint: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1 returned an unexpected response: Unexpected http code: 403```

### CODE REVIEW

File: tests/manual-test-cases/Group21-Registries/21-2-Artifactory.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Purpose:
+To verify that VIC engine works properly with Artifactory
+
+# References:
+
+# Environment:
+This test requires that a vSphere server is running and available
+
+# Test Steps:
+1. Deploy VIC appliance to the vSphere server with
```
Comment (anchal-agrawal) [inline tests/manual-test-cases/Group21-Registries/21-2-Artifactory.md]: ... with `vic-machine`?

File: tests/manual-test-cases/Group21-Registries/21-2-Artifactory.robot [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+*** Test Cases ***
+Test
+    Log To Console  \nStarting test...
+    Install VIC Appliance To Test Server  additional-args=--insecure-registry=vic-docker-local.artifactory.eng.vmware.com
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login -u vic-deployer -p vmware!123 http://vic-docker-local.artifactory.eng.vmware.com/
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Login Succeeded
+    
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} pull vic-docker-local.artifactory.eng.vmware.com/busybox:1
```
Comment (anchal-agrawal) [inline tests/manual-test-cases/Group21-Registries/21-2-Artifactory.robot]: Is `1` the tag of the busybox image on artifactory?

Comment (mhagen-vmware) [reply tests/manual-test-cases/Group21-Registries/21-2-Artifactory.robot]: yep

**Comment by mhagen-vmware** [comment]
Image pull bug:
https://github.com/vmware/vic/issues/6738

**Comment by sflxn** [comment]
lgtm but I guess it's going to be blocked by the pull issue.

**Pull request merged**